.github/scripts/local_flags.py

```diff
@@ -1,4 +1,4 @@
 import os
 
 RUN_LOCAL = not os.environ.get('GITHUB_ACTIONS', False)
```

Comment:
@abejgonzalez is there any reason this should be a separate file from ci_variables? 

It would be nice to move that there and better describe the semantics of local runs vs ci-hosted runs

Reply:
Now that all the `RUN_LOCAL` usages would be in one file (`ci_variables.py`), I think it's fine to delete the `local_flags.py` file. I think there was a circular import issue but now looking at the code I don't see it. I think it would be nice if you could test this change locally by running `.github/scripts/run-mypy-typechecker.py` in your local FireSim clone just to be sure.

Reply:
theres no circular dep, but mypy complains about all manner of things. I'll add the ci scripts directory to the type checking script

Reply:
(in a follow up PR)

Reply:
`mypy` is complaining about non-CI code or CI-code? I haven't added the CI to the typechecking.

Reply:
CI code. Well the non-CI code too, but Tim fixed that.

Reply:
@davidbiancolin are you running mypy with the limited scope that https://github.com/firesim/firesim/blob/c240b2596f67ab6a3a3422565e38ada035715666/.github/scripts/run-mypy-typechecker.py#L11 uses?  You shouldn't see the CI scripts in the output of mypy or you're doing it wrong.

I'd honestly prefer that we switch to using `.mypy.ini` at the root of the repo instead of sticking the "right options" in a wrapper script that I'm always struggling to remember the name of.  However, https://mypy.readthedocs.io/en/stable/config_file.html#:~:text=then%20setup.cfg-,in%20the%20current%20directory,-%2C%20then%20%24XDG_CONFIG_HOME/mypy seems to imply that mypy doesn't look up the tree for a config file if it doesn't find one in the current working directory and I generally dislike that (yet understand why, it's inefficient).

something like https://pre-commit.com/ helps avoid this type of thing because it reduces the number of places stuff gets configured down to one for all the linters instead of one for each checker. (NOTE, I'm also assuming that it normalizes where you run from or at least refuses to run if you're not at the root of the repo but I haven't checked that yet).

Reply:
I modified the bash script to include the ci dirs.

Reply:
Ok.  I was confused by what you were saying and thought you ran mypy directly maybe.